### PR TITLE
[ML][Inference] adds logistic_regression output aggregator

### PR DESCRIPTION
**Pull request by benwtrent**
This adds the logistic_regression output aggregator. 

It takes a set of optional weights (that are in the order of the ensemble's models), multiplies the resultant ensemble sub-models' results, and passes it through a sigmoid function. The return of the sigmoid function is assumed to be the probability of class `1`. Consequently, the probability of class `0` is `1 - p(class_1)`.

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/inference/utils/StatisticsTests.java [hunk 1]
```diff
@@ -30,4 +31,20 @@ public void testSoftMaxWithNoValidValues() {
         expectThrows(IllegalArgumentException.class, () -> Statistics.softMax(values));
     }
 
+    public void testSigmoid() {
+        double eps = 0.000001;
+        List<Tuple<Double, Double>> expectations = Arrays.asList(
```
Comment (przemekwitek) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/inference/utils/StatisticsTests.java]: Maybe rename to `argumentAndExpectedResult` or `inputAndExpectedOutput`? I know it's lengthy but `expectations` does not really tell what this list is about without looking at the assertion below.

Alternatively, you could get rid of this list and rewrite assertions as:
```
assertThat(Statistics.sigmoid(0.0), closeTo(0.5, eps));
```

It's a little bit code duplication but I thinks acceptable in tests.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/ensemble/LogisticRegression.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    }
+
+    public static LogisticRegression fromXContentLenient(XContentParser parser) {
+        return LENIENT_PARSER.apply(parser, null);
+    }
+
+    private final List<Double> weights;
+
+    LogisticRegression() {
+        this.weights = null;
```
Comment (przemekwitek) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/ensemble/LogisticRegression.java]: Alternatively you could call the other constructor using:
```
this((List<Double>) null);
```

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/ensemble/LogisticRegression.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        return this.weights == null ? null : this.weights.size();
+    }
+
+    @Override
+    public List<Double> processValues(List<Double> values) {
+        Objects.requireNonNull(values, "values must not be null");
+        if (weights != null && values.size() != weights.size()) {
+            throw new IllegalArgumentException("values must be the same length as weights.");
+        }
+        double summation = weights == null ?
```
Comment (przemekwitek) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/ensemble/LogisticRegression.java]: This question is probably irrelevant to this particular PR, but why do we even allow `weights` being `null` in aggregators?

Comment (benwtrent) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/ensemble/LogisticRegression.java]: Well defined defaults makes sense for these functions (scikit learn and other ML frameworks do something similar). If the weights are null here, that easily implies a simple summation of the values.

**Review by przemekwitek — APPROVED** [review]
LGTM

**Pull request merged**